tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,269 @@
+    is "mocked", and "call_count" property is introduced and incremented in relevant method for assertions in tests.
+    """
+
+    exclude_field_names: ClassVar[Set[str]] = ParameterBuilder.exclude_field_names | {
+        "call_count",
+    }
+
+    def __init__(
+        self,
+        name: str,
+        evaluation_parameter_builder_configs: Optional[
+            List[ParameterBuilderConfig]
+        ] = None,
+    ) -> None:
+        super().__init__(
+            name=name,
+            evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+            data_context=None,
+        )
+
+        self._call_count: int = 0
+
+    @property
+    def call_count(self) -> int:
+        return self._call_count
+
+    def _build_parameters(
+        self,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> Attributes:
+        """
+        Only "_call_count" is incremented; otherwise, arbitrary test value is returned.
+        """
+        self._call_count += 1
+
+        return Attributes(
+            {
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY: 1,
+                FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY: {"batch_id_0": 1},
+                FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY: {"environment": "test"},
+            }
+        )
+
+
+# noinspection PyUnresolvedReferences,PyUnusedLocal
+@pytest.mark.unit
+@mock.patch(
+    "great_expectations.rule_based_profiler.parameter_builder.parameter_builder.convert_to_json_serializable",
+    return_value="my_json_string",
+)
+def test_parameter_builder_dependencies_evaluated_separately(
+    mock_convert_to_json_serializable: mock.MagicMock,
+):
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=None,
+    )
+
+    domain = DummyDomain()
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_0_parameter_builder.call_count == 1
```

Comment:
I'm still not sure this actually provides much value - asserting that we call a private method doesn't feel like a test that'll prevent errors from sneaking by. What is the actual behavior we want to test? We should be asserting that the built parameters are accurate I think

Reply:
@cdkini To me, this test just has to insure that the dependencies are processed, and calling that implementation method in those dependency ParameterBuilders is necessary and sufficient.  I added a docstring to that effect.  Thanks.